apps/wallet-mobile/src/Staking/PoolTransition/usePoolTransition.tsx

```diff
@@ -0,0 +1,426 @@
+import {PoolInfo} from '@emurgo/yoroi-lib'
+import {useNavigation} from '@react-navigation/native'
+import {useTheme} from '@yoroi/theme'
+import {Balance} from '@yoroi/types'
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {Image, StyleSheet, Text, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import {useQuery} from 'react-query'
+
+import {Button, Icon, Spacer, useModal} from '../../components'
+import {useStakingInfo} from '../../Dashboard/StakePoolInfos'
+import {useSelectedWallet} from '../../features/WalletManager/Context'
+import {formatTimeSpan, Quantities} from '../../yoroi-wallets/utils'
+
+// TODO: PoolTransition and getPoolTransition will come from yoroi-lib
+type PoolTransition = {
+  currentPool: PoolInfo & {tax: number}
+  suggestedPool: PoolInfo & {tax: number}
+  deadlineMilliseconds: number
+}
+
+const getPoolTransition = async (currentPoolId: string): Promise<PoolTransition | null> => {
+  return new Promise((resolve) => {
+    if (currentPoolId === 'd248ded3c18e0e80d07a46f00a2d808075b989ccb1a0e40a76e5cee1') {
+      resolve({
+        currentPool: {
+          pic: 'https://img.cexplorer.io/a/5/5/f/2/pool12eht6dqxpzqj87xuextrpufz2gxmt4reuesuw26r2utzw0kw906.png',
+          ticker: 'EMRG',
+          name: 'emurgo old',
+          roa: '0',
+          tax: '3.5',
+        } as unknown as PoolInfo & {tax: number},
+        suggestedPool: {
+          pic: 'https://img.cexplorer.io/e/c/2/3/1/pool1dkww2vlysa8lsnuf5rca979zdsyr3zvt59hu7e420yxfunkka2z.png',
+          id: 'df1750df9b2df285fcfb50f4740657a18ee3af42727d410c37b86207',
+          name: 'emurgo new',
+          roa: '5.1',
+          ticker: 'EMRG',
+          tax: '3.2',
+        } as unknown as PoolInfo & {tax: number},
+        deadlineMilliseconds: 2999777000,
+      })
+    } else {
+      resolve(null)
+    }
+  })
+}
+
+export const usePoolTransition = () => {
+  const navigation = useNavigation()
+  const wallet = useSelectedWallet()
+  const {stakingInfo} = useStakingInfo(wallet)
+
+  const isStaked = stakingInfo?.status === 'staked'
+  const currentPoolId = isStaked ? stakingInfo?.poolId : ''
+
+  const poolTransitionQuery = useQuery({
+    enabled: isStaked,
+    retry: false,
+    staleTime: 5 * 60 * 1000,
+    queryKey: [wallet.id, 'poolTransition', currentPoolId],
+    queryFn: () => getPoolTransition(currentPoolId),
+  })
+
+  const poolTransition = poolTransitionQuery.data ?? null
```

Comment:
shouldTransition? if pool is retiring?

Reply:
`shouldTransition` sounds like a boolean, and isn't very specific as to what needs transitioning, which is pool.

`poolTransition` existing or being null already signifies the need to transition or not  🤔

Reply:
Added `isPoolRetiring` as a boolean checking it `poolTransition` exists or not.